hoverkite-protocol/src/lib.rs

```diff
@@ -0,0 +1,136 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use core::ops::RangeInclusive;
+
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+pub enum Side {
+    Left,
+    Right,
+}
+
+impl Side {
+    pub fn opposite(self) -> Self {
+        match self {
+            Self::Left => Self::Right,
+            Self::Right => Self::Left,
+        }
+    }
+}
+
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub enum Command {
+    // FIXME: stop using RangeInclusive, so we can derive Copy
+    SetMaxSpeed(RangeInclusive<i16>),
+    SetSpringConstant(u16),
+    SetTarget(i64),
+    Recenter,
+    BatteryReport,
+    Relax,
+    PowerOff,
+}
+
+#[cfg(feature = "std")]
+impl Command {
+    pub fn write_to(&self, mut writer: impl std::io::Write) -> std::io::Result<()> {
+        match self {
+            Command::SetMaxSpeed(max_speed) => {
+                writer.write_all(&[b'S'])?;
+                writer.write_all(&max_speed.start().to_le_bytes())?;
+                writer.write_all(&max_speed.end().to_le_bytes())?;
+            }
+            Command::SetSpringConstant(spring_constant) => {
+                writer.write_all(&[b'K'])?;
+                writer.write_all(&spring_constant.to_le_bytes())?;
+            }
+            Command::SetTarget(target) => {
+                writer.write_all(&[b'T'])?;
+                writer.write_all(&target.to_le_bytes())?;
+            }
+            Command::Recenter => writer.write_all(&[b'e'])?,
+            Command::BatteryReport => writer.write_all(&[b'b'])?,
+            Command::Relax => writer.write_all(&[b'n'])?,
+            Command::PowerOff => writer.write_all(&[b'p'])?,
+        };
+        Ok(())
+    }
+}
+
+// ??? Should we add `Both(Command)`, or add command-specific forwarding
+// ??? logic to the firmware for SetMaxSpeed and SetSpringConstant?
+pub enum DirectedCommand {
```

Comment:
Thinking about it, this comment at code review time is a symptom of the fact that I have only a vague idea of "where could we be?" and I'm trying to forge ahead and work out "how do we get there?" in code without a proper understanding of "where should we be? or "where are we now?".

It would probably be a more efficient use of time to sketch out answers to those questions on a whiteboard together before continuing.